src/index.ts

```diff
@@ -1,15 +1,26 @@
 import * as DCL from 'decentraland-ecs'
-const typesMap = require('decentraland-ecs/types/dcl/decentraland-ecs.api')
 
 type Component = {
   data?: DCL.ObservableComponent
   entityName?: string
 }
 
+interface KeyValuePair<K, T> {
+  key: K
+  value: T
+}
+
 export default class SceneWriter {
+  private DCL
+  private map
   private entities: Map<string, DCL.Entity> = new Map<string, DCL.Entity>()
   private components: Component[] = []
```

Comment:
Components are also a map. Then you have a relation like
```hs
Entities :: Map<EntityID, DCL.Entity>
Components :: Map<ComponentID, Component>
```

Every entity `E` in `Entities` has a `E.components` which is a `Map<ComponentName, Component>`

So.. You may not need to add components to the `SceneWriter` since all components are set in the entities